### DISCUSSION TITLE
Access ComboBox in DataGrid

### DISCUSSION THREAD

**Discussion by ollih1980** [AvaloniaUI/Avalonia#7264] [Q&A]
Hi all,

I've managed to add a ComboBox to my DataGrid, but I'm struggeling to get the selectedItem from it afterwards.

My way to add was:
```
var MyCol = new DataGridTemplateColumn();
                    vCol.CellTemplate = new FuncDataTemplate<object>((itemModel, namescope) =>
                    {
                        return new Avalonia.Controls.ComboBox
                        {
                            Items = {"Test1", "Test2"},
                            BorderThickness = Avalonia.Thickness.Parse("0"),
                            FontSize = 11,
                            HorizontalContentAlignment = Avalonia.Layout.HorizontalAlignment.Stretch,
                            HorizontalAlignment = Avalonia.Layout.HorizontalAlignment.Stretch
                        };
                    });
                    MyDataGrid.Columns.Add(MyCol);
```

Could you pls guide me in how to access this?

Thx.

**Comment by maxkatz6** [comment]
You have your row model inside - itemModel.
You can do whatever you want to synchronize this ComboBox with that model, either you use events or avalonia bindings from code (read documentation).

**Reply by ollih1980** [reply]
What do you mean by saying : "synchronize this ComboBox with that model"??

**Reply by ollih1980** [reply]
@maxkatz6 Could you pls explain a little more??

**Reply by maxkatz6** [reply]
"itemModel" parameter is your row model, right?
Inside of this function that creates new template you can setup events and bindings from the code, as you started doing.
